internal/util/resource/resource.go

```diff
@@ -27,6 +27,10 @@ import (
 	"github.com/FerretDB/FerretDB/v2/internal/util/devbuild"
 )
 
+// cleanupByToken maps each *Token to its runtime.Cleanup handle.
+// That lets Untrack cancel the scheduled cleanup and avoids leaks.
+var cleanupByToken sync.Map
```

Comment:
Do we need both a global map and a token? Can't the token store cleanup function?

Reply:
I was considering violating this comment, 
`
// Token should be a field of a tracked object.
//
// The underlying type is not struct{} because (from the Go spec)
// "Two distinct zero-size variables may have the same address in memory",
// and they do.
` 
but if we create token object as in my last change, we can handle the zero size problem. I've implemented this based on your suggestion.